File: app/src/main/java/no/nav/tilbakemeldingsmottak/rest/bestillingavsamtale/validation/BestillSamtaleValidator.java
package no.nav.tilbakemeldingsmottak.rest.bestillingavsamtale.validation;

import no.nav.tilbakemeldingsmottak.model.BestillSamtaleRequest;
import no.nav.tilbakemeldingsmottak.rest.common.validation.RequestValidator;

public class BestillSamtaleValidator extends RequestValidator {

    public void validateRequest(BestillSamtaleRequest request) {
        hasText(request.getFornavn(), "fornavn");
        hasText(request.getEtternavn(), "etternavn");
        isNotNull(request.getTidsrom(), "tidsrom");
        hasText(request.getTelefonnummer(), "telefonnummer");
    }

}

File: app/src/main/kotlin/no/nav/tilbakemeldingsmottak/rest/bestillingavsamtale/validation/BestillSamtaleValidator.kt
package no.nav.tilbakemeldingsmottak.rest.bestillingavsamtale.validation

import no.nav.tilbakemeldingsmottak.model.BestillSamtaleRequest
import no.nav.tilbakemeldingsmottak.rest.common.validation.RequestValidator

class BestillSamtaleValidator : RequestValidator() {
    fun validateRequest(request: BestillSamtaleRequest) {
        hasText(request.fornavn, "fornavn")
        hasText(request.etternavn, "etternavn")
        isNotNull(request.tidsrom, "tidsrom")
        hasText(request.telefonnummer, "telefonnummer")
        isLegalTelephoneNumber(request.telefonnummer, "telefonnummer")
    }
}
